core/src/main/web/app/helpers/helper.js

```diff
@@ -163,7 +163,27 @@
             }
             bkUtils.httpPost('rest/file-io/setWorkingDirectory', {dir: path});
             // Format this accordingly!
-            bkHelper.openWindow(bkUtils.getBaseUrl() + '/open?uri=' + path, 'notebook');
+            var routeParams = {
+              uri: path
+            }
+            if (uriType) {
+              routeParams.type = uriType;
+            }
+            if (readOnly) {
+              routeParams.readOnly = true;
+            }
+            if (format) {
+              routeParams.format = format;
+            }
+            var serialize = function(obj) {
```

Comment:
I am almost sure we already have something like this function. Anyone know? Is there a cleaner alternative? Or should I just throw this in bkUtils?